src/ProjectSystemTools/BuildLogging/Model/Backend/BackendBuildTableDataSource.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.Providers;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model.Backend
+{
+
+    [Export(typeof(IBackendBuildTableDataSource))]
+    internal sealed class BackendBuildTableDataSource : IBackendBuildTableDataSource // ITableEntriesSnapshotFactory, 
+    {
+        private const string BuildDataSourceDisplayName = "Build Data Source";
+        private const string BuildTableDataSourceIdentifier = nameof(BuildTableDataSourceIdentifier);
+        private const string BuildTableDataSourceSourceTypeIdentifier = nameof(BuildTableDataSourceSourceTypeIdentifier);
+
+        private readonly EvaluationLogger _evaluationLogger;
+        private readonly RoslynLogger _roslynLogger;
+
+        private ImmutableList<Build> _entries = ImmutableList<Build>.Empty;
+
+        public string SourceTypeIdentifier => BuildTableDataSourceSourceTypeIdentifier;
+
+        public string Identifier => BuildTableDataSourceIdentifier;
+
+        public string DisplayName => BuildDataSourceDisplayName;
+
+        public bool SupportRoslynLogging => _roslynLogger.Supported;
+
+        public bool IsLogging { get; private set; }
+
+        public bool SupportsRoslynLogging => _roslynLogger.Supported;
+
+        private NotifyCallback NotifyUI { get; set; }
+
+        public BackendBuildTableDataSource()
+        {
+            _evaluationLogger = new EvaluationLogger(this);
+            _roslynLogger = new RoslynLogger(this);
+        }
+
+        public void Start(NotifyCallback notifyCallback)
+        {
+            NotifyUI = notifyCallback;
+
+            IsLogging = true;
+            ProjectCollection.GlobalProjectCollection.RegisterLogger(_evaluationLogger);
+            _roslynLogger.Start();
+        }
+
+        public void Stop()
+        {
+            NotifyUI = null;
+
+            IsLogging = false;
+            ProjectCollection.GlobalProjectCollection.UnregisterAllLoggers();
+            _roslynLogger.Stop();
+        }
+
+        public void Clear()
+        {
+            foreach (var build in _entries)
+            {
+                build.Dispose();
+            }
+            _entries = ImmutableList<Build>.Empty;
+        }
+
+        public ILogger CreateLogger(bool isDesignTime) => new ProjectLogger(this, isDesignTime);
+
+        /// <summary>
+        /// (Temporary) return log path on server for a given build
```

Comment:
What do you mean by Temporary?

Reply:
Right now this returns the file path to the log file (which would be on the server)
This won't work outside of a local context since the file will need to be transferred across machines.
I'll most likely change this once I start adding async code